### PR TITLE
#185172092 A seller should be able to view a single item and have a different view of all Items in seller's collection

### PR DESCRIPTION
**Pull request by Kress2000**
#### What does this PR do?
This PR enables A seller to view the list of his items , and to view also each specific item and its details

#### Description of Task to be completed?
If a user has logged in as a seller, he has rights to view his items that was previously added, and view each items and its details.

#### How should this be manually tested?
just via Front-End hosted link, you can view it live, and log in as a seller and then view the items you have added

#### What are the relevant pivotal tracker/Trello stories?
#185172092 

#### Screenshots (if appropriate)
am just implementing what i have designed in Figma: 
image
image

### CODE REVIEW

File: src/hooks/useFetch.test.ts
[the remark this answers is not in the document]
Comment (Kress2000) [reply src/hooks/useFetch.test.ts:1]: i had to test hooks too

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+./src/redux
+**/*.js
+*.js
+**/*.css
+*.css
+**/*.scss
+*.scss
+**/*.test.tsx
+*.test.tsx
```
[the remark this answers is not in the document]
Comment (Kress2000) [reply .eslintignore:9]: yes sure let me remove them, thanks for correcting

File: vite.config.ts [hunk 2]
```diff
@@ -16,7 +16,18 @@ export default defineConfig({
...
+        '**/*.stories.{ts,tsx}',
+        '**/components/seller/components/sellerItems/*.tsx',
+        '**/*.slice.{ts,tsx}',
+        '**/main.tsx',
+        '**/hooks.ts',
+        '**/routes/*.{ts,tsx}',
+        '**/redux/*.{ts,tsx}',
+        '**/hooks/*.{ts,tsx}',
+        '**/redux/**/*.{ts,tsx}',
+      ],
```
Comment (Edcreation) [inline vite.config.ts:30]: Why did you exclude all these files?

Comment (Kress2000) [reply vite.config.ts:30]: they are not tested

File: src/components/seller/components/sellerItems/sellerItems.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  totalPages: number;
+}
+
+export default function SellerItems() {
+  const dispatch = useDispatch();
+  //redux get initial values from redux
+  const search = useSelector((state: RootState) => state.filterCollectionProducts.search);
+  const page = useSelector((state: RootState) => state.filterCollectionProducts.page);
+  //get instant data from db
+  const url = `https://team-sostene-e-commerce-bn-production-f4c1.up.railway.app/products/all?page=${page}&limit=10`;
```
Comment (Edcreation) [inline src/components/seller/components/sellerItems/sellerItems.tsx:40]: Use env variables here

**Review by Edcreation — CHANGES_REQUESTED** [review]

File: src/components/seller/components/sellerItems/sellerItems.tsx (hunk 3 above)
Comment (Kress2000) [reply src/components/seller/components/sellerItems/sellerItems.tsx:40]: this receives parameters plz, so it dynamic

Comment (munezerobagira) [reply src/components/seller/components/sellerItems/sellerItems.tsx:40]: what do mean you can use the environment variable

**Review by munezerobagira — APPROVED** [review]

File: src/components/seller/components/sellerItems/sellerItems.tsx (hunk 3 above)
Comment (Edcreation) [reply src/components/seller/components/sellerItems/sellerItems.tsx:40]: https://team-sostene-e-commerce-bn-production-f4c1.up.railway.app this url might change

**Review by Edcreation — APPROVED** [review]

**Review by Pacific911 — APPROVED** [review]
rebase is needed

**Review by munezerobagira — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**